test/stableTRIStaking/migrate.test.ts

```diff
@@ -44,6 +39,32 @@ describe("RevenueDistributionToken - Migrate", function () {
     expect(await this.pTRI.balanceOf(this.deployer.address)).to.equal("100");
   });
 
+  it("should migrate, migrate again and claim initial", async function () {
+    await this.rewardToken.transfer(this.pTRI.address, "100");
+    await this.tri.mint(this.deployer.address, "100");
+    await this.tri.approve(this.xTRI.address, "100");
+    await this.xTRI.enter("100");
+
+    expect(await this.xTRI.balanceOf(this.deployer.address)).to.equal("200");
+
+    await this.pTRI.connect(this.deployer).migrate(this.xTRI.address, "100");
+
+    expect(await this.tri.balanceOf(this.deployer.address)).to.equal("0");
+    expect(await this.xTRI.balanceOf(this.deployer.address)).to.equal("100");
+
+    expect(await this.tri.balanceOf(this.pTRI.address)).to.equal("100");
+    expect(await this.pTRI.balanceOf(this.deployer.address)).to.equal("100");
+
+    await this.xTRI.approve(this.pTRI.address, "100");
+    await this.pTRI.connect(this.deployer).migrate(this.xTRI.address, "100");
+
+    expect(await this.tri.balanceOf(this.pTRI.address)).to.equal("200");
+    expect(await this.pTRI.balanceOf(this.deployer.address)).to.equal("200");
+
+    expect(await this.rewardToken.balanceOf(this.deployer.address)).to.equal("100");
```

Comment:
this would have been claimed in the initial migrate
can you add a reward transfer after initial migrate, and see if it claims again?